test/parallel_api/numeric/numeric.ops/reduce_by_segment.pass.cpp

```diff
@@ -360,17 +358,17 @@ main()
 #endif // TEST_DPCPP_BACKEND_PRESENT
 
 #if !_PSTL_RED_BY_SEG_WINDOWS_COMPILE_ORDER_BROKEN
-    run_test<MatrixPoint<float>, UserBinaryPredicate<MatrixPoint<float>>, MaxFunctor<MatrixPoint<float>>>();
+    run_test</*use_device_alloc=*/true, MatrixPoint<float>, UserBinaryPredicate<MatrixPoint<float>>, MaxFunctor<MatrixPoint<float>>>();
```

Comment:
Yes, every test is alternated with testing USM shared vs. USM device memory instead of testing both. Prior to doing this, the data section of the executable grew far too large resulting in a linker error, " relocation truncated to fit...".

From looking the error message up, this could be avoided by using special compilation flags to use 64-bit relative addressing, but I think trimming the test down a bit makes the most sense.